test/tokenizer.test.ts

```diff
@@ -1,106 +0,0 @@
-      /*identifier*/
-      // identifier identifier
-      /* identifier */
-      `
-      expect(extractFormulaIdentifiers(formula)).toHaveLength(0)
-    })
-  })
-  describe('when there are identifiers', () => {
-    it('should extract a simple identifier', () => {
-      expect(extractFormulaIdentifiers('SomeField__c')).toEqual(['SomeField__c'])
-      expect(extractFormulaIdentifiers(' SomeField__c ')).toEqual(['SomeField__c'])
-    })
-    it('should extract identifier when it`s a function parameter', () => {
-      expect(extractFormulaIdentifiers('ISBLANK(SomeField__c)')).toEqual(['SomeField__c'])
-      expect(extractFormulaIdentifiers('ISBLANK ( SomeField__c )')).toEqual(['SomeField__c'])
-      expect(extractFormulaIdentifiers('ISBLANK(SomeField__c, SomeOtherField)')).toEqual(['SomeField__c', 'SomeOtherField'])
-    })
-    it('should extract identifier when it`s part of arithmetic expression', () => {
-      expect(extractFormulaIdentifiers('"blah" & SomeField__c')).toEqual(['SomeField__c'])
-      expect(extractFormulaIdentifiers('SomeField__c * 0.2')).toEqual(['SomeField__c'])
-    })
-    it('should extract identifier when its name contains a keyword', () => {
-      expect(extractFormulaIdentifiers('trueField && TruE0')).toEqual(['trueField', 'TruE0'])
-      expect(extractFormulaIdentifiers('maybeFalse__c')).toEqual(['maybeFalse__c'])
-    })
-  })
-  describe('when there are identifiers with complex names', () => {
-    it('should extract a simple identifier', () => {
-      expect(extractFormulaIdentifiers('$SomeField__c')).toEqual(['$SomeField__c'])
-      expect(extractFormulaIdentifiers('[SomeField__c]')).toEqual(['[SomeField__c]'])
-      expect(extractFormulaIdentifiers('[SomeField__c].Field1.Field2')).toEqual(['[SomeField__c].Field1.Field2'])
-      expect(extractFormulaIdentifiers('$SomeField__c.Field1.Field2')).toEqual(['$SomeField__c.Field1.Field2'])
-    })
-    it('should extract identifier when it`s a function parameter', () => {
-      expect(extractFormulaIdentifiers('ISBLANK($SomeField__c.Field1)')).toEqual(['$SomeField__c.Field1'])
-      expect(extractFormulaIdentifiers('ISBLANK([SomeField__c].Field1, $SomeOtherField.Field2.Field3)'))
-        .toEqual(['[SomeField__c].Field1', '$SomeOtherField.Field2.Field3'])
-    })
-    it('should extract identifier when it`s part of arithmetic expression', () => {
-      expect(extractFormulaIdentifiers('"blah" & $SomeField__c.Field1')).toEqual(['$SomeField__c.Field1'])
-      expect(extractFormulaIdentifiers('[SomeField__c].Field2 * 0.2')).toEqual(['[SomeField__c].Field2'])
-    })
-  })
-  describe('when there`s more than one identifier', () => {
-    it('Should only extract unique identifiers', () => {
-      expect(extractFormulaIdentifiers('IF(SomeField__c, SomeField__c, SomeField__c)')).toEqual(['SomeField__c'])
-    })
-    it('Should not reorder identifiers', () => {
-      expect(extractFormulaIdentifiers('IF(zzz, zzz, bbb)')).toEqual(['zzz', 'bbb'])
```

Comment:
What happens if the formula is `IF(zzz, bbb, zzz)`?

Reply:
https://github.com/salto-io/salesforce-formula-parser/pull/4